scorecard/scoring.py

```diff
@@ -1,39 +1,122 @@
+    max_technology_score = len(security_capabilities) * get_max_status_value()
+    for product_security_capability in product_security_capabilities:
+        technology_score += product_security_capability.status.value
+
+    # Calculate Compliance score.
+    compliance_score = 0
     status_values = get_control_status_values()
     # Get the product controls, but filter out all those with a status of 'not applicable'.
     product_controls = ProductControl.objects.filter(Q(product=product) & ~Q(status='not applicable')).\
         order_by('control__name')
-    max_score = len(product_controls) * 3  # A maximum possible score is 3 x the number of controls.
-    score = 0
+    # max_compliance_score = len(product_controls) * 3  # A maximum possible score is 3 x the number of controls.
+    max_compliance_score = 1
     for product_control in product_controls:
         try:
-            score += status_values[product_control.status]
+            compliance_score += status_values[product_control.status]
         except KeyError:
             print(f">>>> ERROR: Unknown product control status: '{product_control.status}'")
-    return score, max_score
+
+    return {'people_score':         people_score,
+            'max_people_score':     max_people_score,
+            'process_score':        process_score,
+            'max_process_score':    max_process_score,
+            'technology_score':     technology_score,
+            'max_technology_score': max_technology_score,
+            'compliance_score':     compliance_score,
+            'max_compliance_score': max_compliance_score}
 
 
-def update_product_scores():
+def update_all_product_scores():
     products = Product.objects.all()
     for product in products:
-        score, max_score = get_product_score(product)
-        product.score = score
-        product.max_score = max_score
-        product.percent_score = round((score / max_score) * 100, 1)
-        product.save()
+        update_product_score(product)
+
+
+def update_product_score(product):
+    scores = get_product_scores(product)
+    product.people_score = scores['people_score']
+    product.max_people_score = scores['max_people_score']
+    product.people_percent_score = round(product.people_score * 100 / product.max_people_score, 1)
+    product.process_score = scores['process_score']
+    product.max_process_score = scores['max_process_score']
+    product.process_percent_score = round(product.process_score * 100 / product.max_process_score, 1)
+    product.technology_score = scores['technology_score']
+    product.max_technology_score = scores['max_technology_score']
+    product.technology_percent_score = round(product.technology_score * 100 / product.max_technology_score, 1)
+    product.compliance_score = scores['compliance_score']
+    product.max_compliance_score = scores['max_compliance_score']
+    product.compliance_percent_score = round(product.compliance_score * 100 / product.max_compliance_score, 1)
```

Comment:
You're repeating yourself a lot here. If there might be different types of scores added in the future it might be worthwhile pulling these three attributes into a separate class and then linking it, something like:
```
class Score():
    score = 1
    max = 5

    def percent(self):
        return round(self.score * 100 / self.max, 1)

product.people_score = <score>
product.process_score = <score>
product.technology_score = <score>
product.compliance_score = <score>
```